Core/src/render/Scene.cpp

```diff
@@ -1,32 +1,31 @@
 #include "Scene.h"
+#include "utility\Log.h"
+#include <cassert>
 #include <stdexcept>
 #include <unordered_set>
 
 namespace ojgl {
 
 Scene::Scene(std::shared_ptr<Buffer> buffer, timer::ms_t duration)
-    : _mainBuffer(buffer)
+    : _mainBuffer(std::move(buffer))
     , _duration(duration)
 {
     for (auto& b : buffers()) {
-        if (b != _mainBuffer)
+        if (b != _mainBuffer) {
             b->generateFBO();
+        }
     }
 }
 
-Scene::~Scene()
-{
-}
-
 Buffer& Scene::operator[](const std::string& name)
 {
     for (auto& b : buffers()) {
         if (b->name() == name) {
             return *b;
         }
     }
-
-    //TODO buffer not found, maybe throw runtime error. Yes error would be good I think.
+    LOG_ERROR("Buffer " << name << " not found");
+    assert(false);
```

Comment:
I don't really like the assert here, `exit` from `<cstdlib>` seems a bit better. (We use `exit(1)` in `Window.cpp`)